src/scenic/core/scenarios.py

```diff
@@ -533,6 +533,8 @@ def generateDefaultRequirements(self):
         for obj in filter(
             lambda x: x.requireVisible and x is not self.egoObject, self.objects
         ):
+            if not self.egoObject:
+                raise InvalidScenarioError("requireVisible set to true with no ego.")
```

Comment:
Better add a test covering this. Also I think `requireVisible set to true but no ego is defined` would be clearer.